translator/src/translator.rs

```diff
@@ -0,0 +1,450 @@
+            Expression::List(l) => self.visit_list(l),
+            Expression::Tuple(t) => self.visit_tuple(t),
+            Expression::Dict(d) => self.visit_dict(d),
+            Expression::Set(s) => self.visit_set(s),
+            Expression::Name(n) => self.visit_name(n),
+            Expression::BoolOp(b) => self.visit_bool_op(b),
+            Expression::UnaryOp(u) => self.visit_unary_op(u),
+            Expression::BinOp(b) => self.visit_bin_op(b),
+            Expression::NamedExpr(n) => self.visit_named_expr(n),
+            Expression::Yield(y) => self.visit_yield(y),
+            Expression::YieldFrom(y) => self.visit_yield_from(y),
+            Expression::Starred(s) => self.visit_starred(s),
+            Expression::Generator(g) => self.visit_generator(g),
+            Expression::ListComp(l) => self.visit_list_comp(l),
+            Expression::SetComp(s) => self.visit_set_comp(s),
+            Expression::DictComp(d) => self.visit_dict_comp(d),
+            Expression::Attribute(a) => self.visit_attribute(a),
+            Expression::Subscript(s) => self.visit_subscript(s),
+            Expression::Slice(s) => self.visit_slice(s),
+            Expression::Call(c) => self.visit_call(c),
+            Expression::Await(a) => self.visit_await(a),
+            Expression::Compare(c) => self.visit_compare(c),
+            Expression::Lambda(l) => self.visit_lambda(l),
+            Expression::IfExp(i) => self.visit_if_exp(i),
+            Expression::JoinedStr(j) => self.visit_joined_str(j),
+            Expression::FormattedValue(f) => self.visit_formatted_value(f),
+        }
+    }
+
+    fn visit_constant(&mut self, constant: &Constant) {
+        match constant.value {
+            ConstantValue::None => write!(self.output, "None"),
+            ConstantValue::Ellipsis => write!(self.output, "..."),
+            ConstantValue::Bool(_) => write!(self.output, "bool"),
+            ConstantValue::Str(_) => write!(self.output, "\"{}\"", constant.get_value(&self.file.source).to_string()),
+            ConstantValue::Bytes => write!(self.output, "bytes"),
+            ConstantValue::Tuple => write!(self.output, "tuple"),
+            ConstantValue::Int => write!(self.output, "{}", constant.get_value(&self.file.source).to_string()),
+            ConstantValue::Float => write!(self.output, "{}", constant.get_value(&self.file.source).to_string()),
+            ConstantValue::Complex => write!(self.output, "complex"),
+            /*
+            Constant::Tuple(elements) => {
+                let tuple_elements: Vec<String> = elements
+                    .iter()
+                    .map(|elem| self.translate_constant(elem))
+                    .collect::<Result<Vec<String>, _>>()?;
+                Ok(format!("({})", tuple_elements.join(", ")))
+            },
+            */
+        };
+    }
+
+    fn visit_import(&mut self, import: &Import) {
+        for name in import.names.iter() {
+            if name.name == "torch" {
```

Comment:
maybe useful to have a set of modules that are publicly available(vs. defined locally)

Reply:
Haha yes this is a hack that only works for pytorch right now and will need to be re-thought. We will probably have some C++ implementations for various Python modules, maybe some of them can be generated by this compiler, others (in particular ones with lots of C extension code) it would probably make more sense to write them in C++ directly :)